MetaMask/metamask-extension
# Update design tokens library from 1.5 to 1.6 WIP

## Explanation

Updates the design-token library from `1.5.1` to `1.6.5` as well as related style updates.

Two main updates are included in this version bump:

### **1. Updating warning.default color and deprecating secondary.default**

![Screen Shot 2022-05-18 at 2 24 48 PM](https://user-images.githubusercontent.com/8112138/169159046-8b9f865a-1d38-42f5-8cc7-7fea59420316.png)

Warning.default was updated from yellow to orange to be legible and more accessible as standalone text. As secondary color category has been only used for warning semantic we are deprecating it and using warning.default in secondary.default stead.

- `var(--color-secondary-default)` => `var(--color-warning-default)`
- Adding deprecation tag to color docs in storybook

### **2. Updates icon.default color and adds icon.alternative**

![Screen Shot 2022-05-18 at 2 29 47 PM](https://user-images.githubusercontent.com/8112138/169159079-244a50f1-17cf-41f8-ad90-aabc9414c2d3.png)

icon.default color was updated to a darker grey. icon.alternative was added as a new icon color.

- Reserving `var(--color-icon-default)` for mostly standalone icon buttons such as X(close) or <(back)
- Updating any icons that aren't standalone buttons to use `var(--color-icon-alternative)` 

## Screenshots/Screencaps

Have added appropriate screenshots next to the code changes in the comments

## Manual Testing Steps
Have added screenshots as much as possible for any color updates as there are many.
You could also check the new stories in storybook by
- Going to the latest build of storybook in the PR
- In search bar search for:
  - `Color`
  - `ConfirmDetailRow`
  - `ConnectedAccountsList`
  - `Spinner`

## Pre-Merge Checklist

- [X] ~**IF** this PR fixes a bug, a test that _would have_ caught the bug has been added~ NA
- [X] ~PR is linked to the appropriate GitHub issue~ NA
- [X] ~PR has been added to the appropriate release Milestone~ NA


**References**
design-token changelog https://github.com/MetaMask/design-tokens/blob/main/CHANGELOG.md#160
